### PR TITLE
ADO pipeline for building and deploying website

### LINKED ISSUE #9269: Move website deployment to ADO
We need to move the website deployment pipeline from GitHub Actions to ADO.

### PR DESCRIPTION
**Pull request by tylerbutler**
_This replaces #9265._

Fixes #9269.

There were several old pipeline configs that weren't used so I removed them as well.

### CODE REVIEW

File: tools/pipelines/build-docs.yml [hunk 1]
```diff
@@ -6,11 +6,20 @@
 
 name: $(Build.BuildId)
 
+parameters:
+  - name: deployOverride
```
Comment (tylerbutler) [inline tools/pipelines/build-docs.yml:10]: Enables deployment overrides, similar to how we can override publishing packages.

File: tools/pipelines/build-docs.yml [hunk 2]
```diff
@@ -6,11 +6,20 @@
...
+      - default
+      - skip
+      - force
+
 variables:
-  - name: skipComponentGovernanceDetection
-    value: true
   - group: doc-versions
   - group: storage-vars
+  - group: InfoSec-SecurityResults
```
Comment (tylerbutler) [inline tools/pipelines/build-docs.yml:22]: This isn't used at the moment but will be in a future PR.

File: tools/pipelines/build-docs.yml [hunk 3]
```diff
@@ -105,22 +120,65 @@ stages:
...
-  displayName: GitHub Action
-  pool: server
+- stage: site_build
+  displayName: 'Build website'
   dependsOn: json
   jobs:
-    - job: trigger_github_action
-      displayName: 'Trigger repository_dispatch'
+    - job: build_site
+      condition: in(stageDependencies.json.upload_json.result, 'Succeeded', 'SucceededWithIssues', 'Skipped')
```
Comment (tylerbutler) [inline tools/pipelines/build-docs.yml:128]:
```suggestion
      condition: in(stageDependencies.json.upload_json.result, 'Succeeded', 'SucceededWithIssues')
```

File: tools/pipelines/build-docs.yml [hunk 4]
```diff
@@ -26,12 +35,20 @@ variables:
     value: ${{ eq(variables['Build.SourceBranchName'], variables['RELEASE_BRANCH']) }}
   - name: validRun
     value: ${{ or(variables.releasePipeline, variables.n1Pipeline, variables.latestPipeline) }}
```
Comment (heliocliu) [inline tools/pipelines/build-docs.yml:37]: (not really related) are the `releasePipeline` or `n1Pipeline` variables still available? looks like they were defined in some of the ymls that got deleted, but i'm not sure if they're being provided elsewhere

**Review by heliocliu — APPROVED** [review]

File: tools/pipelines/build-docs.yml (hunk 4 above)
Comment (tylerbutler) [reply tools/pipelines/build-docs.yml:37]: The variables are still defined in ADO, so I left these in place for now.

**Pull request merged**